diff --git a/lua/ui/message.lua b/lua/ui/message.lua
index 149aeb5..0177dd7 100644
--- a/lua/ui/message.lua
+++ b/lua/ui/message.lua
@@ -351,7 +351,7 @@ message.__confirm = function (obj)
 			relative = "editor",
 
 			row = config.row or math.ceil((vim.o.lines - #lines) / 2),
-			col = config.col or math.ceil((vim.o.columns - utils.max_len(lines)) / 2),
+			col = 1,
 
 			width = config.width or utils.max_len(lines),
 			height = config.height or utils.wrapped_height(lines, config.width),
diff --git a/lua/ui/spec.lua b/lua/ui/spec.lua
index 0168a2f..44d4c9a 100644
--- a/lua/ui/spec.lua
+++ b/lua/ui/spec.lua
@@ -8,7 +8,7 @@ spec.default = {
 	popupmenu = {
 		---|fS
 
-		enable = true,
+		enable = false,
 
 		tooltip = function ()
 			local mode = vim.api.nvim_get_mode().mode;
